src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -427,18 +428,21 @@ private static ConfigurationItemFactory BuildDefaultFactory()
                 var allAssemblies = LogFactory.CurrentAppDomain.GetAssemblies();
                 foreach (var assembly in allAssemblies)
                 {
-                    if (assembly.FullName.StartsWith("NLog.", StringComparison.OrdinalIgnoreCase))
+                    if (assembly.FullName.StartsWith("NLog", StringComparison.OrdinalIgnoreCase))
                     {
                         if (!alreadyRegistered.Contains(assembly.FullName))
                         {
                             factory.RegisterItemsFromAssembly(assembly);
                         }
                     }
 
-                    if ( assembly.FullName.StartsWith("NLog.Extensions.Logging", StringComparison.OrdinalIgnoreCase)
-                      || assembly.FullName.StartsWith("NLog.Web", StringComparison.OrdinalIgnoreCase)
-                      || assembly.FullName.StartsWith("Microsoft.Extensions.Logging", StringComparison.OrdinalIgnoreCase)
-                      || assembly.FullName.StartsWith("Microsoft.Logging", StringComparison.OrdinalIgnoreCase))
+                    if ( assembly.FullName.StartsWith("NLog.Extensions.Logging,", StringComparison.OrdinalIgnoreCase)
+                      || assembly.FullName.StartsWith("NLog.Web,", StringComparison.OrdinalIgnoreCase)
+                      || assembly.FullName.StartsWith("NLog.Web.AspNetCore,", StringComparison.OrdinalIgnoreCase)
+                      || assembly.FullName.StartsWith("Microsoft.Extensions.Logging,", StringComparison.OrdinalIgnoreCase)
+                      || assembly.FullName.StartsWith("Microsoft.Extensions.Logging.Abstractions,", StringComparison.OrdinalIgnoreCase)
+                      || assembly.FullName.StartsWith("Microsoft.Extensions.Logging.Filter,", StringComparison.OrdinalIgnoreCase)
+                      || assembly.FullName.StartsWith("Microsoft.Logging,", StringComparison.OrdinalIgnoreCase))
                     {
```

Comment:
Why the duplicates, it's a "starts with"

Reply:
Notice the `,`

Reply:
Ah thanks, hardly to see on mobile.